### PR TITLE
BG-1399: Stripe status handling for manual bank verification

### PR DESCRIPTION
**Pull request by jp-mariano**
Resolves BG-1399

## Explanation of the solution
When a donor opts in for manual bank verification, Stripe will send microdeposits into the nominated bank account. This deposit may appear within days so we would have to make sure that the bank verification link is saved to a pending transaction or at least give the option to copy the link for guest donors. Stripe also sends an email containing instructions on how to manually verify their bank account but it's better if we also keep track of this type of events.

A logged in donor should be able to see in their My Donations page's "Pending" tab a new row that contains a "Verify bank account" button in the "Action" row.

## Instructions on making this work
- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
- donate `USD` via bank account and choose manual bank verification to simulate a `requires_action` event from Stripe
  - verify that the links are accessible in Donation thank you and My Donations pages
- donate `USD` via bank account and choose the readily available "Test Institution" to simulate retry mechanism in new version of API
  - should arrive finally at Donation thank you page

### Donation thank you page (logged in donor)
Screenshot 2024-06-26 at 12 05 46

### Donation thank you page (guest donor)
Screenshot 2024-06-26 at 12 43 35

### Pending tab on My Donations page for tx that requires manual verification
Screenshot 2024-06-28 at 12 25 16

### CODE REVIEW

File: src/pages/UserDashboard/Donations/Table.tsx [hunk 1]
```diff
@@ -79,6 +79,9 @@ export default function Table({
             USD Value
           </HeaderButton>
           {status === "intent" ? <></> : <>TX Hash</>}
+          {status === "pending" && (
```
Comment (ap-justin) [inline src/pages/UserDashboard/Donations/Table.tsx:82]: @jp-mariano  @SovereignAndrey, perhaps col name should be more generic like `Actions` 
image

as other donations may not be `bank`, in this case `Action` could just be `- - - `

**Review by ap-justin — CHANGES_REQUESTED** [review]

File: src/services/apes/apes.ts [hunk 2]
```diff
@@ -125,11 +125,13 @@ export const apes = createApi({
         guestDonor?: GuestDonor;
         recipientName?: string;
         recipientId?: number;
+        arrivalDate?: number;
```
[the remark this answers is not in the document]
Comment (jp-mariano) [reply src/services/apes/apes.ts]: Addressed in commit https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/3107/commits/4c7c71013689f679be528377dc8927405c056fd3.

File: src/types/aws/ap/donations.ts [hunk 3]
```diff
@@ -41,6 +41,9 @@ export type DonationRecord = {
   initAmountUsd?: number;
   finalAmountUsd?: number;
   splitLiqPct: number;
+
+  // bank verification url
```
[the remark this answers is not in the document]
Comment (jp-mariano) [reply src/types/aws/ap/donations.ts]: Addressed in commit https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/3107/commits/51a49068138e2de048aa53dd847485559ae1a1ec.

File: src/pages/UserDashboard/Donations/Table.tsx (hunk 1 above)
Comment (jp-mariano) [reply src/pages/UserDashboard/Donations/Table.tsx:82]: Addressed in commit https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/3107/commits/d375820a1fa43d14f9bf704dfbb6782271fde22d. Also updated PR's description copy.

**Review by ap-justin — APPROVED** [review]

**Review by SovereignAndrey — APPROVED** [review]

**Pull request merged**